sql/connect/server/src/test/scala/org/apache/spark/sql/connect/execution/ReattachableExecuteSuite.scala

```diff
@@ -429,4 +429,21 @@ class ReattachableExecuteSuite extends SparkConnectServerTest {
     val abandonedExecutions = manager.listAbandonedExecutions
     assert(abandonedExecutions.forall(_.operationId != dummyOpId))
   }
+
+  test("SPARK-49492: reattach succeeds on an inactive execution holder") {
+    val dummyOpId = UUID.randomUUID().toString
+    val dummyRequest =
+      buildExecutePlanRequest(buildPlan("select * from range(1)"), operationId = dummyOpId)
+    val manager = SparkConnectService.executionManager
+    val holder = manager.createExecuteHolder(dummyRequest)
+    holder.eventsManager.postStarted()
+    manager.removeExecuteHolder(holder.key)
+    withRawBlockingStub { stub =>
+      val reattach = stub.reattachExecute(buildReattachExecuteRequest(dummyOpId, None))
+      val e = intercept[StatusRuntimeException] {
+        reattach.hasNext()
+      }
+      assert(e.getMessage.contains("INVALID_HANDLE.OPERATION_NOT_FOUND"))
+    }
+  }
```

Comment:
I'm not sure if this is testing the right thing here as it would pass regardless of the changes proposed in `SparkConnectExecutePlanHandler` IIUC. 

Is this meant more of a general test to verify that we return OPERATION_NOT_FOUND when the execute holder is removed?

If that's the case, could we add another test which would mimic hitting an exception in the try block introduced above and verify the `OPERATION_NOT_FOUND` error?

Essentially, the test would look like:

1. Create a dummy `proto.ExecutePlanRequest`
2. Call `SparkConnectExecutePlanHandler#handle` on the request
3. Have an exception thrown in the try block (perhaps via mocking / overriding one of the operations in the block)
4. Attempt to reattach to the above execution
5. verify op_not_found error

Reply:
Thanks, I'll try that!